internal/connectors/engine/workflow/create_payout.go

```diff
@@ -52,6 +52,32 @@ func (w Workflow) createPayout(
 		return err
 	}
 
+	// If the payout is scheduled in the future, we need to add a schedule
+	// for processing adjustment to the payment initiation, and then sleep until
+	// the scheduled time.
+	now := workflow.Now(ctx)
+	if !pi.ScheduledAt.IsZero() && pi.ScheduledAt.After(now) {
+		err = w.addPIAdjustment(
+			ctx,
+			models.PaymentInitiationAdjustmentID{
+				PaymentInitiationID: createPayout.PaymentInitiationID,
+				CreatedAt:           workflow.Now(ctx),
+				Status:              models.PAYMENT_INITIATION_ADJUSTMENT_STATUS_SCHEDULED_FOR_PROCESSING,
+			},
+			pi.Amount,
+			&pi.Asset,
+			nil,
+			map[string]string{
+				"scheduledAt": pi.ScheduledAt.String(),
+			},
+		)
+		if err != nil {
+			return err
+		}
+
+		workflow.Sleep(ctx, pi.ScheduledAt.Sub(now))
```

Comment:
Can you check if using a sleep instead of a start delay will cause our metrics to get skewed? I don't think it'll be great if we start counting the sleep time as part of the workflow execution time metric since a single transaction planned to the future could skew the whole average of the dataset.

Reply:
We have the activity latency metric as well which is more indicative of how long things take to process, so I think this should be fine.